sdk/openai/openai/test/public/chatCompletions.spec.ts

```diff
@@ -18,21 +21,23 @@ import {
 } from "../utils/asserts.js";
 import { z } from "zod";
 import { zodResponseFormat } from "openai/helpers/zod";
-import { type ChatCompletionMessageParam } from "openai/resources/chat/completions.mjs";
-import { functionCallModelsToSkip, jsonResponseModelsToSkip } from "../utils/models.js";
+import type { ChatCompletionMessageParam } from "openai/resources/chat/completions.mjs";
+import { functionCallModelsToSkip } from "../utils/models.js";
 import "../../src/types/index.js";
-import type { ClientsAndDeploymentsInfo } from "../utils/types.js";
 import { assertMathResponseOutput, type MathResponse } from "../utils/structuredOutputUtils.js";
 
 describe.shuffle.each(APIMatrix)("Chat Completions [%s]", (apiVersion: APIVersion) => {
-  let clientsAndDeploymentsInfo: ClientsAndDeploymentsInfo;
-
-  clientsAndDeploymentsInfo = createClientsAndDeployments(
+  const clientsAndDeploymentsInfo = createClientsAndDeployments(
     apiVersion,
     { chatCompletion: "true" },
     {
+      clientOptions: { maxRetries: 0 },
```

Comment:
Retries is a major source of slowness of our tests. I think it is better to skip to the next deployment rather than waiting for a full minute for the current one. This PR also updates deployment filtering to pick ones with highest capacity, which greatly reduces the risk of rate limiting errors.